src/Symfony/Component/DependencyInjection/Tests/Fixtures/xml/services10.xml

```diff
@@ -11,6 +11,10 @@
             other-option="lorem"
             an_other-option="ipsum"
         />
+        <tag some-option="cat"
+            some_option="ciz"
+            other-option="lorem"
+            an_other-option="ipsum">bar_tag</tag>
```

Comment:
```suggestion
            an_other-option="ipsum">bar_tag
        </tag>
```

Reply:
The chance cause that the tag have a newline with some whitespace, and when the container extract the services with his tag the tags does not match
![image](https://user-images.githubusercontent.com/32807382/205516700-a8de0598-8232-4126-8a4e-ef3798834538.png)
and cause the test to not pass
![image](https://user-images.githubusercontent.com/32807382/205516738-8eba97d3-c773-436f-820e-a6c14357ad33.png)

Reply:
Could you add `trim()` by replacing:
```php
'' === $tagName
```
with:
```php
'' === trim($tagName)
```
?

Reply:
I can not use a function in a write context
![image](https://user-images.githubusercontent.com/32807382/205550612-5b3e3d3d-c84a-4326-a679-ca77258a8c58.png)

I need to `trim()` the content before the assigment, to look something like this:
```php
if ('' === $tagName = trim((0 !== $tag->childElementCount || '' === $tag->nodeValue) ? $tag->getAttribute('name') : $tag->nodeValue)) {
    throw new InvalidArgumentException(sprintf('The tag name for service "%s" in "%s" must be a non-empty string.', (string) $service->getAttribute('id'), $file));
}
```

is it ok?

Reply:
Yes